optaplanner-core/src/main/java/org/optaplanner/core/impl/solver/DefaultSolver.java

```diff
@@ -135,12 +140,34 @@ public boolean isTerminateEarly() {
 
     @Override
     public boolean addProblemFactChange(ProblemFactChange<Solution_> problemFactChange) {
-        return basicPlumbingTermination.addProblemFactChange(problemFactChange);
+        return basicPlumbingTermination.addProblemChange(new ProblemFactChangeAdapterImpl<>(problemFactChange));
     }
 
     @Override
     public boolean addProblemFactChanges(List<ProblemFactChange<Solution_>> problemFactChangeList) {
-        return basicPlumbingTermination.addProblemFactChanges(problemFactChangeList);
+        Objects.requireNonNull(problemFactChangeList, "The list of problem fact changes ("
+                + problemFactChangeList + ") cannot be null.");
+        List<ProblemChangeAdapter<Solution_>> adaptedProblemChangeList = problemFactChangeList.stream()
+                .map(ProblemFactChangeAdapterImpl::new)
+                .collect(Collectors.toList());
+        return basicPlumbingTermination.addProblemChanges(adaptedProblemChangeList);
```

Comment:
I think you missed one of the list conversions.

Reply:
Also, you could reduce code duplication a bit - instead of calling the termination, you can call the other `addProblemFactChange(...)` and that will do the conversion and termination call for you.

The same for the non-deprecated methods.

Reply:
Not really missed; the problem here is we must return the `boolean`, and in the Javadoc, we say "Follows specifications of {@link BlockingQueue#addAll(Collection)}".